clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -1539,30 +1543,30 @@ public void testOffsetOfPausedPartitions() {
 
         client.requests().clear();
         consumer.unsubscribe();
-        consumer.close();
+        consumer.close(Duration.ZERO);
     }
 
     @Test
     public void testPollWithNoSubscription() {
-        try (KafkaConsumer<byte[], byte[]> consumer = newConsumer((String) null)) {
-            assertThrows(IllegalStateException.class, () -> consumer.poll(Duration.ZERO));
-        }
+        KafkaConsumer<byte[], byte[]> consumer = newConsumer((String) null);
+        assertThrows(IllegalStateException.class, () -> consumer.poll(Duration.ZERO));
+        consumer.close(Duration.ZERO);
     }
 
     @Test
     public void testPollWithEmptySubscription() {
-        try (KafkaConsumer<byte[], byte[]> consumer = newConsumer(groupId)) {
-            consumer.subscribe(Collections.emptyList());
-            assertThrows(IllegalStateException.class, () -> consumer.poll(Duration.ZERO));
-        }
+        KafkaConsumer<byte[], byte[]> consumer = newConsumer(groupId);
+        consumer.subscribe(Collections.emptyList());
+        assertThrows(IllegalStateException.class, () -> consumer.poll(Duration.ZERO));
+        consumer.close(Duration.ZERO);
     }
 
     @Test
     public void testPollWithEmptyUserAssignment() {
-        try (KafkaConsumer<byte[], byte[]> consumer = newConsumer(groupId)) {
```

Comment:
Are we loosing anything by removing all those try resources? It basically means that the consumer is not closed in case of an exception.

Reply:
yes. try-with-resources is calling `KafkaConsumer.close()` which has a timeout set to 30s. These tests will wait for default timeout to exit (close() in the tests is expected to be unsuccessful because there is no server to send the request to) and hence, unnecessarily, increase the execution time of these tests.